src/custom-element.js

```diff
@@ -1040,33 +1191,41 @@ function createBaseCustomElementClass(win, elementConnectedCallback) {
         this.tryUpgrade_();
         return;
       }
-      if (this.buildingPromise_) {
-        // Already building.
+
+      if (this.mountPromise_) {
+        // Already mounting.
         return;
       }
 
-      // Schedule build.
-      this.setReadyStateInternal(ReadyState.BUILDING);
-      const builder = getBuilderForDoc(this.getAmpDoc());
-      builder.schedule(this);
+      // Schedule build and mount.
+      const scheduler = getSchedulerForDoc(this.getAmpDoc());
+      scheduler.schedule(this);
 
-      // Schedule preconnects.
-      const urls = this.implClass_.getPreconnects(this);
-      if (urls && urls.length > 0) {
-        // If we do early preconnects we delay them a bit. This is kind of
-        // an unfortunate trade off, but it seems faster, because the DOM
-        // operations themselves are not free and might delay
-        const ampdoc = this.getAmpDoc();
-        startupChunk(ampdoc, () => {
-          const {win} = ampdoc;
-          if (!win) {
-            return;
-          }
-          const preconnect = Services.preconnectFor(win);
-          urls.forEach((url) =>
-            preconnect.url(ampdoc, url, /* alsoConnecting */ false)
-          );
-        });
+      if (this.buildingPromise_) {
```

Comment:
I'm confused as to whether we need `buildingPromise` or `mountingPromise`.

Reply:
The `mountPromise` is already checked above. I clarified the comment a bit more. But basically in this case the building is already known to have started, so the element must be in the "mounting" phase. Does this make sense?

Reply:
Ahh, ok.